### PR TITLE
Allow all members of private orgs to see public repos (#11442)

### LINKED ISSUE #10144: Public repositories within private org are not visible to members
- Gitea version (or commit ref): 1.10.3 / 1.12.0+dev-251-gd4096ab6a
- Git version: Not relevant
- Operating system: Not relevant
- Database:
  - [ ] PostgreSQL
  - [x] MySQL
  - [ ] MSSQL
  - [ ] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [x] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
- Log gist: Not relevant

## Description

When repository, within an organization which has visibility set to Private, is set as public and user is part of a team that has no access specified to this repository, repository can be accessed via direct URI but can not be viewed on any list within UI.

I would expect that setting repository as public within Private organization would allow members of all teams to view it, regardless of team's permissions and team's repositories (for example I might want to make repository visible but read-only to all members of organization, while certain teams can write to different repositories).

This does not apply when organization is set as Limited. In such a case, repository is properly visible (to both members of organization as well as outside users).

Owners of org can of course see all repositories just fine.

## Screenshots

chrome_2020-02-05_01-32-09
chrome_2020-02-05_01-32-25
chrome_2020-02-05_01-32-38
chrome_2020-02-05_01-32-46
chrome_2020-02-05_01-32-52
firefox_2020-02-05_01-31-20
firefox_2020-02-05_01-31-33
firefox_2020-02-05_01-31-57

### PR DESCRIPTION
**Pull request by zeripath**
Backport (#11442)

Allow all members of private orgs to see public repos

Fix #10144

Signed-off-by: Andrew Thornton <art27@cantab.net>

### CODE REVIEW

**Review by lafriks — APPROVED** [review]

File: models/repo_list.go [hunk 1]
```diff
@@ -340,41 +340,37 @@ func SearchRepositoryByCondition(opts *SearchRepoOptions, cond builder.Cond) (Re
 // accessibleRepositoryCondition takes a user a returns a condition for checking if a repository is accessible
 func accessibleRepositoryCondition(userID int64) builder.Cond {
 	if userID <= 0 {
+		// Public repositories that are not in private or limited organizatons
```
Comment (lafriks) [inline models/repo_list.go]:
```suggestion
		// Public repositories that are not in private or limited organizations
```

**Comment by lafriks** [comment]
This breaks tests

**Comment by zeripath** [comment]
Sighbiscuits.

**Review by guillep2k — APPROVED** [review]

**Comment by guillep2k** [comment]
Ping LG-TM

**Review by lafriks — APPROVED** [review]

**Pull request merged**